cdap-error-api/src/main/java/io/cdap/cdap/error/api/ScopedException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+package io.cdap.cdap.error.api;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+public class ScopedException extends Exception implements ErrorTagProvider {
```

Comment:
Javadocs

Reply:
Consider making it a `RuntimeException`. This is a helper intermediate exception. It won't be very convenient as a checked one

Reply:
Removed from this PR.